### PR TITLE
feat: add `web-ext` for development and pack

### LINKED ISSUE #27: Content scripts in dev mode don't work in Firefox
Great work on this project! But since I mainly develop in FF I thought I would mention that there is an issue specific to FF which works fine in chromium browsers. If dev mode is enabled (HMR) the content scripts don't even load in the page. Build works fine in FF. Without ```delete manifest.content_scripts``` the development mode works, but requires the extension to be reloaded manually.

### PR DESCRIPTION
**Pull request by tshemeng**
Features

- add `start:chromium` for live coding with Chromium
- add `start:firefox` for live coding with Firefox, fix #27
- add `pack` for build `.crx` and `.xpi` extension both

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -93,7 +94,19 @@ pnpm i
...
 ```
 
-Then **load extension in browser with the `extension/` folder**,
+Then **load extension in browser with the `extension/` folder**.
+
+For firefox extension developers, just load `extension/` folder to browser,
+
+contentScript hmr not work as Chromium based browser.
+
+As alternative, you can run firefox with command:
```
Comment (tmkx) [inline README.md:103]: These should be one line.

File: README.md [hunk 2]
```diff
@@ -93,7 +94,19 @@ pnpm i
...
+
+```bash
+pnpm run start:firefox
```
Comment (tmkx) [inline README.md:106]: just `pnpm start:firefox`

File: README.md [hunk 3]
```diff
@@ -103,7 +116,9 @@ To build the extension, run
 pnpm build
 ```
 
-And then pack files under `extension`.
+And then pack files under `extension`,
+
+you can upload `extension.crx` or `extension.xpi` to appropriate extension store.
```
Comment (tmkx) [inline README.md:121]: in one line, plz

File: package.json [hunk 4]
```diff
@@ -13,7 +13,11 @@
     "build:prepare": "esno scripts/prepare.ts",
     "build:web": "vite build",
     "build:js": "vite build --config vite.config.content.ts",
-    "pack": "crx pack extension -o ./extension.crx",
+    "pack": "cross-env NODE_ENV=production run-s run-p pack:*",
+    "pack:crx": "crx pack extension -o ./extension.crx",
```
Comment (tmkx) [inline package.json:17]: It seems that `web-ext` supports `chromium-based` browsers, can we use it to replace `crx pack`? Have you tested and verified this situation?

https://extensionworkshop.com/documentation/develop/web-ext-command-reference/#:~:text=Here%20are%20the%20supported%20targets

Comment (tshemeng) [reply package.json:17]: `web-ext build` currently only supports output `.xpi` and `.zip` fromat.

File: README.md (hunk 3 above)
Comment (tshemeng) [reply README.md]: fixed

File: README.md [hunk 5]
```diff
@@ -93,7 +94,17 @@ pnpm i
 pnpm dev
 ```
 
-Then **load extension in browser with the `extension/` folder**,
+Then **load extension in browser with the `extension/` folder**.
+
+For firefox extension developers, just load `extension/` folder to browser, contentScript hmr not work as Chromium based browser.
+
+As alternative, you can run firefox with command:
```
Comment (antfu) [inline README.md:101]:
```suggestion
For Firefox developers, you can run the following command instead:
```

**Review by antfu — APPROVED** [review]
Please resolve the conflict and feel free to merge this by your own. Thanks.

**Comment by tshemeng** [comment]
feat: add web-ext for development and pack

**Pull request merged**